node/src/Pos/Block/Worker.hs

```diff
@@ -97,11 +97,21 @@ blkOnNewSlot =
 -- Block creation worker
 ----------------------------------------------------------------------------
 
-blockCreator
-    :: WorkMode ssc ctx m
-    => SlotId -> SendActions m -> m ()
+blockCreator ::
+       forall ssc ctx m. WorkMode ssc ctx m
+    => SlotId
+    -> SendActions m
+    -> m ()
 blockCreator (slotId@SlotId {..}) sendActions = do
 
+    -- Print difference between tip slot and current slot.
+    -- TODO: probably not the best place, but it's not the biggest concern now.
+    tipHeader <- DB.getTipHeader @ssc
+    let tipSlot = epochOrSlotToSlot (getEpochOrSlot tipHeader)
+    let slotDiff = flattenSlotId slotId - flattenSlotId tipSlot
```

Comment:
@Martoon-00 Silly question: are we assuming that `slotId` will be the greater of the two or should `tipSlot` be? In other terms, are we expecting `slotDiff` to yield negative values, in case `slotId < tipSlot`?

Reply:
AFAIU `slotId` here is id of slot which has just began, judging by current time. It can be smaller than `tipSlot` only if we get blocks "from future", which perhaps may happen only if local time is set incorrectly, but this seems unlikely

Reply:
Gotcha, sounds fair enough!